variables.tf

```diff
@@ -4,6 +4,14 @@ variable "name" {
   description = "Name used for resources and depending resources."
 }
 
+variable "resource_specific_tags" {
+  description = "Map of tags to assign to specific resources supporting tags. Merged with `tags`."
```

Comment:
Let's list the supported object types, or use the new optional object attributes right away (https://www.terraform.io/docs/language/expressions/type-constraints.html#experimental-optional-object-type-attributes)